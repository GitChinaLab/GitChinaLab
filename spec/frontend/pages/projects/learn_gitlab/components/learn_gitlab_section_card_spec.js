import { shallowMount } from '@vue/test-utils';
import LearnGitlabSectionCard from '~/pages/projects/learn_gitlab/components/learn_gitlab_section_card.vue';
import { testActions } from './mock_data';

const defaultSection = 'workspace';
const testImage = 'workspace.svg';

describe('Learn GitLab Section Card', () => {
  let wrapper;

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  const createWrapper = () => {
    wrapper = shallowMount(LearnGitlabSectionCard, {
      propsData: { section: defaultSection, actions: testActions, svg: testImage },
    });
  };

  it('renders correctly', () => {
    createWrapper({ completed: false });

    expect(wrapper.element).toMatchSnapshot();
  });
});
