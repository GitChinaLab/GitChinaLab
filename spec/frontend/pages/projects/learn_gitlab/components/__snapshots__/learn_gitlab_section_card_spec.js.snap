// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Learn GitLab Section Card renders correctly 1`] = `
<gl-card-stub
  bodyclass=""
  class="gl-pt-0 learn-gitlab-section-card"
  footerclass=""
  headerclass=""
>
  <div
    class="learn-gitlab-section-card-header"
  >
    <img
      src="workspace.svg"
    />
     
    <h2
      class="gl-font-lg gl-mb-3"
    >
      Set up your workspace
    </h2>
     
    <p
      class="gl-text-gray-700 gl-mb-6"
    >
      Complete these tasks first so you can enjoy GitLab's features to their fullest:
    </p>
  </div>
   
  <learn-gitlab-section-link-stub
    action="userAdded"
    value="[object Object]"
  />
  <learn-gitlab-section-link-stub
    action="issueCreated"
    value="[object Object]"
  />
  <learn-gitlab-section-link-stub
    action="gitWrite"
    value="[object Object]"
  />
  <learn-gitlab-section-link-stub
    action="mergeRequestCreated"
    value="[object Object]"
  />
  <learn-gitlab-section-link-stub
    action="securityScanEnabled"
    value="[object Object]"
  />
  <learn-gitlab-section-link-stub
    action="pipelineCreated"
    value="[object Object]"
  />
  <learn-gitlab-section-link-stub
    action="trialStarted"
    value="[object Object]"
  />
  <learn-gitlab-section-link-stub
    action="codeOwnersEnabled"
    value="[object Object]"
  />
  <learn-gitlab-section-link-stub
    action="requiredMrApprovalsEnabled"
    value="[object Object]"
  />
</gl-card-stub>
`;
